### PR TITLE
return trigger result

### LINKED ISSUE #4793: V7: Trigger validation with return
**Is your feature request related to a problem? Please describe.**
We have a blocker before we can upgrade to v7. We currently have some code that looks like this:

```javascript
const onClick = async () => {
  const valid = await inputForm.trigger();
  if (!valid) {
    return;
  }
  const data = inputForm.getValues();
  // use data
}
```

We are currently unable to move to v7 since I cannot figure out how to do this in v7.

**Describe the solution you'd like**
Some way to trigger validation and get the result back.

**Describe alternatives you've considered**
None

### LINKED ISSUE #4903: Errors properties off by one render outside render
**Describe the bug**
When we need to do something after the validation (in another function) is done, it seems like `errors`-properties are off by one render on their values.

**To Reproduce**

1. Open console (to see output)
2. Go to firstName input-field
3. Remove John
4. Insert a single space-character ' '
5. Click outside input-field (to trigger onBlur)
6. Look at console-output
7. Go to firstName input-field
8. Remove the single space and leave input blank
9. Click outside input-field (to trigger onBlur)
10. Look at console-output

**Codesandbox link (Required)**
https://codesandbox.io/s/small-thunder-nq0t4

**Expected behavior**
Expected that `errors.firstName` contains the error as the same as `error` does in the console.

**Screenshots**
image
(Image above) This one shows that `errors` contains the correct error, but `errors.firstName` is undefined. Should not say `Do something`. The UI-error is correct.

image
(Image above) This one shows that `errors` contains the correct error, but `errors.firstName` is `errors` previous value. Should say `Don't do something`. The UI-error is correct.

**Additional context**
Have tried the same in JS (not TS) same result.
Tested on version: 7.2.2 and 7.1.1

### PR DESCRIPTION
**Pull request by bluebill1049**
close #4793 
close #4903

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -451,18 +451,20 @@ export function useForm<
           ),
         );
       } else {
-        isUndefined(name)
+        isValid = !!(isUndefined(name)
           ? await validateForm(fieldsRef.current)
           : await Promise.all(
               fields.map(async (data) => await executeValidation(data, null)),
-            );
+            ));
```
Comment (KirchoKirev) [inline src/useForm.ts:458]: @bluebill1049 Is this correct? `await Promise.all(...)` will return an array with boolean values which is always truthy. If we want the correct value for `isValid` in this case we would want to do `(await Promise.all(...)).every(Boolean)` just like in [v6](https://github.com/react-hook-form/react-hook-form/blob/v6.15.7/src/useForm.ts#L429)?

Comment (bluebill1049) [reply src/useForm.ts:458]: oh wow, good pick! want to send a PR, credit goes to you.

Comment (bluebill1049) [reply src/useForm.ts:458]: I will wait till end of today, hopefully you can make it and should be another minor release.

Comment (KirchoKirev) [reply src/useForm.ts:458]: Hey, I saw that you already made the change, thank you :)

Comment (bluebill1049) [reply src/useForm.ts:458]: no worries! thank you for point this out!

**Pull request merged**